### ISSUE TITLE
Support scheduled contest pause

### ISSUE THREAD

**Issue by prabowo02** [ia-toki/judgels#227]
As a manager, I want to have the "Paused" module enabled and disabled automatically at a certain time.

Useful when you want to pause your contest during Pelatnas lunchtime and unpause it again once the lunchtime is over.

**Comment by lungsin** [comment]
Design Proposal: 

1.  In addition to the current pause module, there will be config to specify the schedule.
2. The user can specify multiple windows in the config. Each window consists of two attributes: the pause start time and the pause duration. 
3. To support the legacy pause module where the user can pause by toggling, the config should support a mode where the contest is paused indefinitely. A simple checkbox in the config will do.
4. The default config is to have the indefinite pause mode on (checkbox at point 3), with no scheduled window (at point 2). The default value for each window is open to discussion.

**Comment by fushar** [comment]
Yes, that looks good :) Implementation of multiple window configs in React would be challenging :D

**Comment by aguss787** [comment]
Rather than having specific scheduling for some modules, how about we create a simple IFTTT module? I don't think the cost will be much higher, while providing us with more flexibility for future development.

**Comment by fushar** [comment]
What is a "simple IFTTT module"? :)

**Comment by aguss787** [comment]
An "if this than that", an event driven action executor.

For example, if "current time is 11.00" then "enable pause module". IIRC, there are some use cases of such module in TOKI scientific committee such as setting contest start time daily, etc.

**Comment by fushar** [comment]
It's possible, but implementation will be a bit tricky because then we need to have a scheduler that runs periodically to trigger the action transition.

**Comment by aguss787** [comment]
Yup, that's true. I think we can make use of dropwizard's quartz to emit an event every minute or so, and then have another worker that will consume that event and do the appropriate task
